lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp

```diff
@@ -307,767 +299,6 @@ FunctionPass *llvm::createScalarReplAggregatesHLSLPass(bool UseDomTree, bool Pro
   return new SROA_SSAUp_HLSL(Promote);
 }
 
-//===----------------------------------------------------------------------===//
```

Comment:
I sneaked in a removal of all of this dead code. We're very far diverged from the original `ScalarReplAggregates.cpp` already and even that code has been deprecated in newer LLVM versions in favor of `SROA.cpp`.